python/link_prediction.py

```diff
@@ -0,0 +1,970 @@
+import mgp  # Python API
+import torch
+import dgl  # geometric deep learning
+from dgl import AddSelfLoop
+from typing import Callable, List, Tuple, Dict
+from numpy import int32
+from dataclasses import dataclass, field
+from collections import defaultdict
+from heapq import heappop, heappush 
+from mage.link_prediction import (
+    preprocess,
+    inner_train,
+    inner_predict,
+    create_model,
+    create_optimizer,
+    create_predictor,
+    GRAPH_SAGE,
+    GRAPH_ATTN,
+    ADAM_OPT,
+    SGD_OPT,
+    CUDA_DEVICE,
+    CPU_DEVICE,
+    DOT_PREDICTOR,
+    MLP_PREDICTOR,
+    MEAN_AGG,
+    LSTM_AGG,
+    POOL_AGG,
+    GCN_AGG,
+    HIDDEN_FEATURES_SIZE,
+    LAYER_TYPE,
+    NUM_EPOCHS,
+    OPTIMIZER,
+    LEARNING_RATE,
+    SPLIT_RATIO,
+    NODE_FEATURES_PROPERTY,
+    DEVICE_TYPE,
+    CONSOLE_LOG_FREQ,
+    CHECKPOINT_FREQ,
+    AGGREGATOR,
+    METRICS,
+    LAYER_TYPE,
+    PREDICTOR_TYPE,
+    ATTN_NUM_HEADS,
+    TR_ACC_PATIENCE,
+    MODEL_SAVE_PATH,
+    CONTEXT_SAVE_DIR,
+    TARGET_RELATION,
+    NUM_NEG_PER_POS_EDGE,
+    BATCH_SIZE,
+    SAMPLING_WORKERS,
+    MODEL_NAME,
+    PREDICTOR_NAME,
+    AUC_SCORE,
+    ACCURACY,
+    PRECISION,
+    RECALL,
+    POS_EXAMPLES,
+    NEG_EXAMPLES,
+    POS_PRED_EXAMPLES,
+    NEG_PRED_EXAMPLES,
+    LOSS,
+    F1,
```

Comment:
Constants can be hidden behind some class. These can be the properties for class values.